### PR TITLE
Fix LICENSE link

### LINKED ISSUE #11135: MIT licence: dead link
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.14--2024-04-02--748b0ee Linux 6.1.0-18-amd64 amd64  Java 21.0.2  JavaFX 22+30

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Go to Help -> About JabRef
2. Click on "MIT"
The URL https://github.com/JabRef/jabref/blob/main/LICENSE.md opens in my browser, reading "404 - page not found".

In fact, the file is here: https://github.com/JabRef/jabref/blob/main/LICENSE  (without the .md extension).

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### PR DESCRIPTION
**Pull request by subhramit**
Closes #11135 
This is a small fix to the link to JabRef's MIT license file, as pointed out in the issue. The license file is LICENSE and not LICENSE.md
### Mandatory checks

- ~[ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)~
- ~[ ] Tests created for changes (if applicable)~
- [x] Manually tested changed features in running JabRef (always required)
- ~[ ] Screenshots added in PR description (for UI changes)~
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by subhramit** [comment]
I pulled from JabRef/jabref (main) before working with this. Why does it still show 4 files changed?

**Comment by Siedlerchr** [comment]
you commited the submoduel changes as well, intellij does this by default

**Comment by subhramit** [comment]
> you commited the submoduel changes as well, intellij does this by default

What is the way to prevent this?

**Comment by subhramit** [comment]
Hi, the issue has been closed before my PR getting merged. Just informing, as I don't think anyone else's fix was merged.

**Comment by HoussemNasri** [comment]
@Siedlerchr Maybe we should change `ignore = all` to `ignore = dirty` in .gitsubmodules? It says in the [documentation](https://git-scm.com/docs/gitmodules) that using ` ignore = all` would include the changes of the submodules in they were staged. With dirty, only committed changes are included.

**Comment by koppor** [comment]
> @Siedlerchr Maybe we should change `ignore = all` to `ignore = dirty` in .gitsubmodules? It says in the [documentation](https://git-scm.com/docs/gitmodules) that using ` ignore = all` would include the changes of the submodules if they were staged. With dirty, only committed changes are included.

@HoussemNasri Sure! Just submit a PR and we'll merge and try...

**Comment by HoussemNasri** [comment]
Hey @subhramit, you need to rebase your branch on top of the upstream main so that your local submodules point to the same commit as the remote submodules. This is how to do it:

Define upstream if it's not already defined
```bash
git remote add upstream https://github.com/JabRef/jabref
```

Fetch the latest changes from upstream

```bash
git fetch upstream
```

Rebase your local branch on top of upstream/main

```bash
git rebase upstream/main fix-for-issue-11135
```

Force push changes

```bash
git push --force
```

**Comment by subhramit** [comment]
> ```shell
> git rebase upstream/main fix-for-issue-11135
> ```

Hi Houssem, thanks for the steps. While trying to rebase, this is what I'm getting:
image
How do I fix this?
Do I have to set up the local from scratch again?

**Comment by Siedlerchr** [comment]
Best is do a git reset --hard upstream/main 
Afterwards you need to do your code changes again and create a new commit

**Comment by subhramit** [comment]
I manually replaced the csl-locales and styles as it was showing a version disparity even after hard reset. Merging wasn't happening so I aborted it, and that apparently closed this PR. Nevertheless, opened a new one. Have to learn a lot more about Git, it seems.

**Comment by koppor** [comment]
@subhramit `git gui` is your friend... Also `git submodule update`.

**Comment by HoussemNasri** [comment]
@koppor Please pardon my git ignorance if what I'm saying is nonsense :^) but isn't `git submodule update` what created the problem in the first place? GitHub shows the diff between the dev branch and the main branch, and submodules AFAIK are just pointers to commits in another repository. Since main was pointing to an older commit of the submodules (it's been a while since someone run git submodule update on it), and I'm assuming Subrahmit run `git submodule update`  on the dev branch, this created a difference between the commit the submodules at main were pointing to and the submodules at the dev branch.

**Comment by koppor** [comment]
TL;DR: It is https://youtrack.jetbrains.com/issue/IDEA-285237/ignored-changes-in-submodules-are-still-visible-in-the-commit-window - you should vote so that this will be fixed somehow!

> Does `git pull upstream main` automatically do `git submodule update`?

I don't know. It should not make any impact. The submodules contain the CSL files and the abbreviation lists. There are only minor changes there.

Someone should google for the answer and/or make a minimal setup :)



My driver: I do `git checkout X` on the command line and use `git gui` for committing. With `ignore = dirty`, if have dozens of changes. See https://github.com/JabRef/jabref/pull/11148#issuecomment-2041437440. Therefore, I switched back to `ignore = all` at https://github.com/JabRef/jabref/pull/11161.

I do not use IntelliJ for branch switching or other git-related tasks. It is an IntelliJ issue.

**You all should vote up https://youtrack.jetbrains.com/issue/IDEA-285237/ignored-changes-in-submodules-are-still-visible-in-the-commit-window.**

**Comment by Siedlerchr** [comment]
When commiting you have to uncheck the submodules in the commit/changes dialog as well

**Comment by koppor** [comment]
> When commiting you have to uncheck the submodules in the commit/changes dialog as well 

I think this is what the JetBrains issue is about. Someone of us should provide a screenshot...

**Comment by koppor** [comment]
@HoussemNasri @subhramit I read the youtrack issue more carefully now. 

I bet, your IntelliJ settings show as following:

image

But they should be different. Please try following setting in IntelliJ:

image

As howto:

1. Open preferences
2. Navigate to Version Control -> Directory Mappings
3. Select "buildres/abbrv.jabref.org".
4. Click "-" on the menu bar
5. Repeat steps 3 and 4 for "src/main/resources/csl-locales" and "src/main/resources/csl-styles"
6. Click "Apply"

**Comment by subhramit** [comment]
> @HoussemNasri @subhramit I read the youtrack issue more carefully now. 
> 
> I bet, your IntelliJ settings show as following:
> 
> image
> 
> But they should be different. Please try following setting in IntelliJ:
> 
> image
> 
> As howto:
> 
> 1. Open preferences
> 2. Navigate to Version Control -> Directory Mappings
> 3. Select "buildres/abbrv.jabref.org".
> 4. Click "-" on the menu bar
> 5. Repeat steps 3 and 4 for "src/main/resources/csl-locales" and "src/main/resources/csl-styles"
> 6. Click "Apply"

Coincidentally (or not) there was a small popup that came up regarding this yesterday itself, something mentioning an error about these directories being roots. On clicking the suggested action, it took me to that window where I did "-" on those three and it got resolved. Thanks!
